app/eth2wrap/eth2wrap.go

```diff
@@ -106,6 +115,94 @@ func (m multi) Address() string {
 	return m.clients[0].Address()
 }
 
+func (m multi) SubmitBeaconCommitteeSubscriptionsV2(ctx context.Context, subscriptions []*eth2exp.BeaconCommitteeSubscription) ([]*eth2exp.BeaconCommitteeSubscriptionResponse, error) {
+	const label = "submit_beacon_committee_subscriptions_v2"
+
+	res0, err := provide(ctx, m.clients,
+		func(ctx context.Context, cl Client) ([]*eth2exp.BeaconCommitteeSubscriptionResponse, error) {
+			return cl.SubmitBeaconCommitteeSubscriptionsV2(ctx, subscriptions)
+		},
+		nil,
+	)
+	if err != nil {
+		incError(label)
+		err = errors.Wrap(err, "eth2wrap")
+	}
+
+	return res0, err
+}
+
+func NewWrapHTTP(eth2Svc eth2client.Service) (Client, error) {
```

Comment:
suggest removing this and just using `Wrap` above

Reply:
but `Wrap` shouldn't care about the underlying type to be `eth2http.Service`. `Wrap` should only care about the type should be `eth2Client.Service` and it should implement methods from `eth2wrap.Client`. And httpWrap is wrapper around `eth2http.Service` and not `eth2Client.Service`.

Reply:
well, we only support `eth2http.Service` at this point, no use trying to be "generic" if there is no such thing

Reply:
actually no, we need this "generic" nature to support beaconmock